auto-value-gson/src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java

```diff
@@ -554,9 +540,61 @@ public TypeSpec createTypeAdapter(Context context,
     return methodSpecs;
   }
 
+  private static String typeAdapterGetter(Property prop,
+                                          ClassName jsonAdapter,
+                                          TypeMirror typeAdapterFactory,
+                                          Types typeUtils,
+                                          List<TypeVariableName> typeParams) {
+    TypeName type = prop.type.isPrimitive() ? prop.type.box() : prop.type;
+    ParameterizedTypeName adp = ParameterizedTypeName.get(jsonAdapter, type);
+    CodeBlock.Builder codeBuilder = CodeBlock.builder();      
+
+    if (prop.typeAdapter != null) {
+      if (typeUtils.isAssignable(prop.typeAdapter, typeAdapterFactory)) {
+        if (prop.type instanceof ParameterizedTypeName || prop.type instanceof TypeVariableName) {
+          codeBuilder.add("(($T) new $T().create(gson, $L))", adp,
```

Comment:
Can we ditch the extra parenthesis? They make the generated code a bit more complex to read without adding value.

Reply:
This is an artifact of the previous version where there were no fields and generated sequence was used directly as a method receiver rather than being RHS of the assignment. Will remove in the next revision.